pkg/apiserver/handlers/networkpolicyanalysis/handler.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicyanalysis
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+	"antrea.io/antrea/pkg/controller/networkpolicy"
+)
+
+// ParsePeer parses Pod/Service name and Namespace, nil is returned if the string
+// is not of a valid Pod/Service reference ("Namespace/name").
+func ParsePeer(str string) (string, string) {
+	parts := strings.Split(str, "/")
+	n := len(parts)
+	if n == 2 {
+		if parts[0] == "" {
+			return "default", parts[1]
+		}
+		return parts[0], parts[1]
+	}
+	return "", ""
+}
+
+// higherPriority compares the two rules and return if newRule has a higher priority than oldRule.
+func higherPriority(oldRule, newRule *networkpolicy.RuleInfo) bool {
+	if oldRule == nil {
+		return true
+	}
+	// compare tier priority
+	effectiveTierPriorityK8sNP := (networkpolicy.DefaultTierPriority + networkpolicy.BaselineTierPriority) / 2
+	if oldRule.Policy.TierPriority == nil {
+		oldRule.Policy.TierPriority = &effectiveTierPriorityK8sNP
+	}
+	if newRule.Policy.TierPriority == nil {
+		newRule.Policy.TierPriority = &effectiveTierPriorityK8sNP
+	}
+	if oldRule.Policy.TierPriority != newRule.Policy.TierPriority {
+		return *newRule.Policy.TierPriority > *oldRule.Policy.TierPriority
+	}
+	// compare policy priority if tier priority equals
+	if oldRule.Policy.Priority != nil && newRule.Policy.Priority != nil && *newRule.Policy.Priority != *oldRule.Policy.Priority {
+		return *newRule.Policy.Priority > *oldRule.Policy.Priority
+	}
+	if newRule.Policy.Name != oldRule.Policy.Name {
+		return newRule.Policy.Name > oldRule.Policy.Name
+	}
+	// compare rule priority if policy priority equals
+	if newRule.Policy.Rules[newRule.Index].Priority != oldRule.Policy.Rules[oldRule.Index].Priority {
+		return newRule.Policy.Rules[newRule.Index].Priority > oldRule.Policy.Rules[oldRule.Index].Priority
+	}
+	// compare rule index if rule priority equals
+	return newRule.Index < oldRule.Index
+}
+
+// intersectEndpoints returns the predicted rules effective from srcEndpoints to dstEndpoints.
+// Rules returned satisfies either in source applied policies and destination egress rules,
+// or in source ingress rules and destination applied policies.
+func intersectEndpoints(srcEndpoints, dstEndpoints *networkpolicy.EndpointRuleAnalysis) *networkpolicy.Rule {
```

Comment:
As discussed, this function needs to take K8s NP implicit isolation into account. Also the function name seems a bit misleading.

Reply:
Updated